### PR TITLE
Revert "Revert "[AArch64][AsmParser] Directives should clear transitively implied features (#106625)""

### PR DESCRIPTION
**Pull request by vitalybuka**
Reverts llvm/llvm-project#106813

### CODE REVIEW

File: llvm/lib/Target/AArch64/AsmParser/AArch64AsmParser.cpp [hunk 1]
```diff
@@ -7075,30 +7075,22 @@ bool AArch64AsmParser::parseDirectiveCPU(SMLoc L) {
...
-      setAvailableFeatures(ComputeAvailableFeatures(ToggleFeatures));
-      FoundExtension = true;
+    auto It = llvm::find_if(ExtensionMap, [&Name](const auto &Extension) {
+      return Extension.Name == Name;
+    });
 
-      break;
-    }
+    if (It == std::end(ExtensionMap))
+      Error(CurLoc, "unsupported architectural extension: " + Name);
```
Comment (vitalybuka) [inline llvm/lib/Target/AArch64/AsmParser/AArch64AsmParser.cpp:7083]: We can't continue this loop if `It == std::end(ExtensionMap)`

Comment (vitalybuka) [reply llvm/lib/Target/AArch64/AsmParser/AArch64AsmParser.cpp:7083]: but it fails tests

**Comment by vitalybuka** [comment]
@labrinea This PR is just for visibility.
Please reland with a fix at your convenience.

**Comment by labrinea** [comment]
Raised https://github.com/llvm/llvm-project/pull/106850

**Pull request closed without merging**